### PR TITLE
Creating a new generic Analytics API integration guide using 3scale…

### LINKED ISSUE #410: Update OpenShift.io Analytics Integration API Guide to reflect all changes and improvements made in the API
The following API Guide needs to be updated:
https://docs.openshift.io/analytics-integration-api-guide.html

Issues:
3.2.6. Example response
- the response format, schema, and meaning of values are not described
- some values (`usage_outliers`) are empty so consumers of the API won't know what expect
- special values needs to be described (where to expect `N/A`, -1 etc.]

3.3.6. Example response
- the response format, schema, and meaning of values are not described.

### PR DESCRIPTION
**Pull request by Preeticp**
…for int and ext users.
This PR also updates the swagger specification link in the OSIO API integration doc which fixes #410

### CODE REVIEW

**Comment by fabric8cd** [comment]
@Preeticp snapshot docs image is available. Try `docker pull quay.io/openshiftio/fabric8io-fabric8-online-docs:SNAPSHOT-PR-460-19`, run the image using `docker run -it -p 4000:8080 quay.io/openshiftio/fabric8io-fabric8-online-docs:SNAPSHOT-PR-460-19`. Docs are now available for preview at `http://127.0.0.1:4000` Thanks

File: docs/topics/modules/about_analytics_platform.adoc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[id="about_analytics_platform"]
+= About analytics platform
+
+Analytics is an open source platform aimed primarily at assisting developers in getting insights and recommendations on the stacks used for developing applications. This platform can be used to get an overview and recommendations for single application components as well.
+
+== Supported ecosystems
+
+The analytics _stack-analyses_ currently analyzes all application stacks in the Java (Maven) and Node (npm) ecosystems.
```
Comment (invincibleJai) [inline docs/topics/modules/about_analytics_platform.adoc]: `The analytics _stack-analyses_`  maybe we can add The analytics _stack-analyses_ endpoint or something to be more explicit

File: docs/topics/modules/api_get_component_analyses_response.adoc [hunk 2]
```diff
@@ -6,11 +6,9 @@
 
 Fetches various data points for a selected component.
 
-NOTE: You require an {osio} authentication token to invoke this method.
+== Resource URL and access key
 
-== Resource URL
-
-For information about the resource URL, reach out to us on the {osio} support chat channel at link:https://chat.{osio}[https://chat.openshift.io] or email us at link:mailto:openshiftio@redhat.com[openshiftio@redhat.com].
+You require the 3scale gateway URL and access key to invoke all of the analytics API resources mentioned below. For information about the gateway URL and the gateway access key, reach out to us on the fabric8-analytics support chat channel at link:https://chat.openshift.io[https://chat.openshift.io].
```
Comment (invincibleJai) [inline docs/topics/modules/api_get_component_analyses_response.adoc]: https://chat.openshift.io should be https://chat.openshift.io/developers/channels/fabric8-analytics

File: docs/topics/modules/api_get_component_analyses_response.adoc [hunk 3]
```diff
@@ -26,7 +24,7 @@ None
 == Example request with cURL
 
 ----
-curl -k -H "Authorization: Bearer <openshift.io authentication token>" https://<recommender api url>/api/v1/component-analyses/maven/io.vertx:vertx-core/3.4.1
+curl https://<3scale APICast Gateway URL>/component-analyses/maven/io.vertx:vertx-core/3.4.1?user_key=<user-key>
```
Comment (invincibleJai) [inline docs/topics/modules/api_get_component_analyses_response.adoc]: https://<3scale APICast Gateway URL>/component-analyses/maven/io.vertx:vertx-core/3.4.1?user_key=<user-key>
should be 
https://<3scale APICast Gateway URL>/api/v1/component-analyses/maven/io.vertx:vertx-core/3.4.1?user_key=<user-key>

File: docs/topics/modules/api_get_component_analyses_response.adoc [hunk 4]
```diff
@@ -36,75 +34,68 @@ curl -k -H "Authorization: Bearer <openshift.io authentication token>" https://<
 [source,typescript]
 ----
 {
+    "schema": {
+        "url": "http://recommender.api.openshift.io/api/v1/schemas/api/analyses_graphdb/1-2-0/",
```
Comment (invincibleJai) [inline docs/topics/modules/api_get_component_analyses_response.adoc]: do we need to expose http://recommender.api.openshift.io as we asked for 3scale gaeway url

File: docs/topics/modules/api_post_stack_analyses_request.adoc [hunk 5]
```diff
@@ -6,11 +6,9 @@
 
 Submits an application stack analysis request.
 
-NOTE: You require an {osio} authentication token to invoke this method.
+== Resource URL and access key
 
-== Resource URL
-
-For information about the resource URL, reach out to us on the {osio} support chat channel at link:https://chat.openshift.io[https://chat.openshift.io] or email us at link:mailto:openshiftio@redhat.com[openshiftio@redhat.com].
+You require the 3scale gateway URL and access key to invoke all of the analytics API resources mentioned below. For information about the gateway URL and the gateway access key, reach out to us on the fabric8-analytics support chat channel at link:https://chat.openshift.io[https://chat.openshift.io].
```
Comment (invincibleJai) [inline docs/topics/modules/api_post_stack_analyses_request.adoc]: same as above , should point f8-analytics MM channel

File: docs/topics/modules/getting_3scale_gateway_URL_and_access_key.adoc [hunk 6]
```diff
@@ -0,0 +1,6 @@
+[id="getting_3scale_gateway_URL_and_access_key"]
+= Getting 3scale gateway URL and access key
+
+You require the 3scale gateway URL and access key to invoke all of the analytics API resources mentioned below. For information about the gateway URL and the gateway access key, reach out to us on the fabric8-analytics support chat channel at link:https://chat.openshift.io[https://chat.openshift.io].
```
Comment (invincibleJai) [inline docs/topics/modules/getting_3scale_gateway_URL_and_access_key.adoc]: same as above

File: docs/topics/modules/api_get_component_analyses_response.adoc (hunk 3 above)
Comment (Preeticp) [reply docs/topics/modules/api_get_component_analyses_response.adoc]: @samuzzal-choudhury can you please confirm on this?

File: docs/topics/modules/api_get_component_analyses_response.adoc (hunk 4 above)
Comment (Preeticp) [reply docs/topics/modules/api_get_component_analyses_response.adoc]: @samuzzal-choudhury can you please confirm if this needs to be replaced and if so with what?

File: docs/topics/modules/api_get_component_analyses_response.adoc (hunk 3 above)
Comment (samuzzal-choudhury) [reply docs/topics/modules/api_get_component_analyses_response.adoc]: @Preeticp I confirm. Let's go with @invincibleJai 's suggestion.

File: docs/topics/modules/api_get_component_analyses_response.adoc (hunk 4 above)
Comment (samuzzal-choudhury) [reply docs/topics/modules/api_get_component_analyses_response.adoc]: @Preeticp  Let's replace `http://recommender.api.openshift.io` with `http://<recommender api url>`.

File: docs/topics/modules/api_get_component_analyses_response.adoc (hunk 3 above)
Comment (samuzzal-choudhury) [reply docs/topics/modules/api_get_component_analyses_response.adoc]: @Preeticp  `https://<3scale APICast Gateway URL>/api/v1/component-analyses/maven/io.vertx:vertx-core/3.4.1?user_key=<user-key>`

**Comment by fabric8cd** [comment]
@Preeticp snapshot docs image is available. Try `docker pull quay.io/openshiftio/fabric8io-fabric8-online-docs:SNAPSHOT-PR-460-23`, run the image using `docker run -it -p 4000:8080 quay.io/openshiftio/fabric8io-fabric8-online-docs:SNAPSHOT-PR-460-23`. Docs are now available for preview at `http://127.0.0.1:4000` Thanks

**Comment by fabric8cd** [comment]
@Preeticp snapshot docs image is available. Try `docker pull quay.io/openshiftio/fabric8io-fabric8-online-docs:SNAPSHOT-PR-460-24`, run the image using `docker run -it -p 4000:8080 quay.io/openshiftio/fabric8io-fabric8-online-docs:SNAPSHOT-PR-460-24`. Docs are now available for preview at `http://127.0.0.1:4000` Thanks

**Pull request merged**